pkg/controller/add_cluster.go

```diff
@@ -170,24 +172,61 @@ func (c *Controller) EnsurePlacement(cluster *myspec.M3DBCluster) error {
 	return nil
 }
 
-// EnsureNamespace will retrieve current namespaces to ensure one matches
-// the cluster name or create a new namespace to match the cluster name
-func (c *Controller) EnsureNamespace(cluster *myspec.M3DBCluster) error {
+// EnsureNamespace will retrieve current namespaces to ensure one matches the
+// cluster name or create a new namespace to match the cluster name. Returns a
+// bool indicating whether a namespace was created and any errors encountered.
+func (c *Controller) EnsureNamespace(cluster *myspec.M3DBCluster) (bool, error) {
 	// Get namespace
 	namespaces, err := c.namespaceClient.List()
 	if err != nil {
 		c.logger.Error("failed to get namespace ", zap.Error(err))
-		return err
+		return false, err
 	}
 	for _, md := range namespaces {
 		if md.ID().Equal(ident.StringID(cluster.GetName())) {
 			c.logger.Info("namespace found", zap.String("ns", md.ID().String()))
-			return nil
+			return false, nil
 		}
 	}
 	if err = c.namespaceClient.Create(cluster.GetObjectMeta().GetName()); err != nil {
 		c.logger.Error("failed to create namespace", zap.Error(err))
+		return false, err
+	}
+	return true, nil
+}
+
+func (c *Controller) ensureServices(cluster *myspec.M3DBCluster) error {
+	// TODO(schallert): support updating service spec, not sure if this only
```

Comment:
Yea that is fair because the heuristic is a simple check that the service exist or not against the entire service configuration such as ports.